### ISSUE TITLE
Linux engine crashes at random due to having more than 1024 file handles open and calling FD_SET to buffer overflow

### ISSUE THREAD

**Issue by nick-l-o3de** [o3de/o3de#15332]
**Describe the bug**

Background:  If you need to wait for files to have data and you don't want to busy loop while waiting but would rather sleep and wake up when data is ready, one common approach is to use the system call select().

The select() system call on linux will sleep until any one of the file handles you feed it in a fd_set structure has data ready.  So to sleep until data is ready and wake up when it is,  you use the macro FD_SET() to populate a fd_set struct with the file handles you'd like to wait on, and then invoke select() on that fd_set, telling it to block until one of those file handles in the set has data.  It will stop using CPU on your thread (letting other stuff on the system have that compute power) until those handles are readable, at which point it will wake your thread up and you can read the data.

The FD_SET function, however, is a bitset, and in most linux systems, is hardcoded to have at most room for a file handle with an integer handle of up to 1023.  Example in openbsd:  But linux is similar.
https://github.com/openbsd/src/blob/fba4865f1dbe0cc6c4725437366d812456e9331d/sys/sys/select.h#L58

Using FD_SET(int) and giving an integer > 1024 will cause a buffer overflow, and apparmor/buffer overflow detection will kill your application.

O3DE's streamer settings by default cause it to keep 1024 file handles open.  file handles are globally shared amonst all applications in linux.  This means that not just in O3DE, but any other application that you run after starting O3DE, if that application is using FD_SET and select, that application will buffer overrun and crash!  Every file handle any application has open increases the file handle integer, it is a global resource.

We should fix this in 2 ways
1. futureproof it by using epoll instead of select() so that it wont crash even if a lot of handles are in use
2. reduce the number of handles o3de keeps open by default, perhaps to a reasonable 128 or 256, since this will destabilize other applications on linux if abused.

I have a fix for this, pending PR.  I will link it here. 

**Steps to reproduce**
1. Start O3DE and load a fairly complicated level.  I was using the ROS2 demo.
2. Cause O3DE to launch a child process (such as XACRO, or any other child process, for example, kill AP and let it restart AP for you).  Shader compiler, anything
**Expected behavior**
Everything works

**Actual behavior**
Your entire linux system is destabilized.  random applications that still use the old API will die.  Any part of O3DE that launches a subapplication (like the import robot dialog xacro invoker and waits for output) will crash and take the editor with it.

**Found in Branch**
Stabilization 2023

**Desktop/Device (please complete the following information):**
 - Device: Ubuntu linux

**Comment by nick-l-o3de** [comment]
This was merged into stabilization and then into dev for release, this can be closed.
